htlcswitch/packet.go

```diff
@@ -29,12 +29,27 @@ type htlcPacket struct {
 	src lnwire.ShortChannelID
 
 	// amount is the value of the HTLC that is being created or modified.
-	//
 	// TODO(andrew.shvv) should be removed after introducing sphinx payment.
 	amount btcutil.Amount
 
 	// htlc lnwire message type of which depends on switch request type.
 	htlc lnwire.Message
+
+	// obfuscator is entity which is needed to make the obfuscation of the
```

Comment:
1. As I recall we agreed on storing the `obfuscators` in the switch payment circuits, because in the next `PR` circuit becomes persistent.

2. In order to re-encrypt the errors in the channel link, we should store the `obfuscators` in it too, and make them persistent too.

3. In current implementation switch as well as channel link creates the payment failures, so it means that we should store `obfuscator` in switch and also in channel links.